src/inmanta/data/__init__.py

```diff
@@ -3875,20 +3850,16 @@ async def get_compile_details(cls, environment: uuid.UUID, id: uuid.UUID) -> Opt
             version=requested_compile["version"],
             do_export=requested_compile["do_export"],
             force_update=requested_compile["force_update"],
-            metadata=json.loads(requested_compile["metadata"]) if requested_compile["metadata"] else {},
-            environment_variables=(
-                json.loads(requested_compile["used_environment_variables"])
-                if requested_compile["used_environment_variables"] is not None
-                else {}
-            ),
-            requested_environment_variables=(json.loads(requested_compile["requested_environment_variables"])),
-            mergeable_environment_variables=(json.loads(requested_compile["mergeable_environment_variables"])),
+            metadata=requested_compile["metadata"] or {},
+            environment_variables=requested_compile["used_environment_variables"] or {},
```

Comment:
What made this work transparently?

Reply:
Not too sure yet, I'll have a closer look

Reply:
Got it:
- regular asyncpg doesn't deserialize by default (https://github.com/MagicStack/asyncpg/issues/560)
- sqlalchemy uses json.loads by default for dialects that support the JSON datatype ([docs](https://docs.sqlalchemy.org/en/20/core/engines.html#sqlalchemy.create_engine.params.json_deserializer))

Reply:
and add it to the raw connection